scripts/fast-reboot

```diff
@@ -139,31 +139,19 @@ systemctl stop syncd
 # Warm reboot: dump state to host disk
 # Note: even if syncd changed ASIC_DB before killed, we don't care
 if [[ "$REBOOT_TYPE" = "warm-reboot" ]]; then
-    # Dump redis content to directory
-    # Note: don't use pretty mode redis-dump (1.1) since there is a known bug with key pattern
-    DUMP_CMD="redis-dump -s /var/run/redis/redis.sock"
+    # Dump redis content to a file 'dump.rdb' in warmboot directory
     mkdir -p $WARM_DIR
-    # Note: requiring redis-dump-load
-    # Save applDB in /host/warm-reboot/appl_db.json
-    local pids=()
-    $DUMP_CMD -d 0 -o $WARM_DIR/appl_db.json &
-    pids+=($!)
-    # Save configDB in /host/warm-reboot/config_db.json
-    $DUMP_CMD -d 4 -o $WARM_DIR/config_db.json &
-    pids+=($!)
-    # Save stateDB (only FDB_TABLE and WARM_RESTART_TABLE) in /host/warm-reboot/state_db.json
-    # WARNING WARNING WARNING: a trick to dump both FDB_TABLE|* and WARM_RESTA*
-    # TODO: replace it with readable mechanism to dump multiple key patterns into one single json file
-    $DUMP_CMD -d 6 -k "[FW][DA][BR][_M][T_][AR][BE][LS][ET][|A]*" -o $WARM_DIR/state_db.json &
-    pids+=($!)
-    # Save asicDB in /host/warm-reboot/asic_db.json
-    $DUMP_CMD -d 1 -o $WARM_DIR/asic_db.json &
-    pids+=($!)
-    # Save loglevelDB in /host/warm-reboot/loglevel_db.json
-    $DUMP_CMD -d 3 -o $WARM_DIR/loglevel_db.json &
-    pids+=($!)
-
-    wait ${pids[@]}
+    # Delete keys in stateDB except FDB_TABLE|* and WARM_RESTA*
+    redis-cli -n 6 eval "
+        for _, k in ipairs(redis.call('keys', '*')) do
+            if not string.match(k, 'FDB_TABLE|') and not string.match(k, 'WARM_RESTART_TABLE|') then
+                redis.call('del', k)
+            end
+        end
+    " 0
+    redis-cli save
+    docker cp database:/var/lib/redis/$REDIS_FILE $WARM_DIR
```

Comment:
Can we read the location and file name from redis.conf instead off hard-code value here?

Reply:
/etc/redis/redis.conf may be not the ground truth. I can find it from `redis-cli config get '*'`.
To make life easier, I prefer keep it hard-coded now. I will revisit if later we have requirement to make it flexible.